### DISCUSSION TITLE
Auto-indexing PP in combination w/ NextCloud

### DISCUSSION THREAD

**Discussion by FlyingPersian** [photoprism/photoprism#1023] [Q&A] [answered]
Hello,

I'm running PP in a docker environment on my Raspberry Pi. So far, everything is working fine. I'm also using nextcloud on the same device to sync both documents and my pictures accross multiple devices (PC & Laptop). In order not to have my pictures (15k+) duplicated, I created one shared volume for NC and PP. I changed ownership to the same user for both containers (1000) and the host system. I use the app "Folder Sync" on my phone to sync the pictures to Nextcloud. I make a folder for each year, so I want the folder 2021 to contain only pictures of 2021, 2020 of 2020 etc. Nextcloud itself doesn't allow for filtering this way, which is why I use Folder Sync.

I have a few questions regarding this:

1) Is this recommended? If not, is there a better way?

2) In Issue #281 I read a few ways of automatic the auto-indexing of originals. I'm not uploading anything via PP, so I have to manually update the index as of now to keep it up2date. I figured I could use cron on my host system or docker. The pics folder is only updated once a day (2am), so I could create a cron tab for 3am to make sure NC and PP don't interfere with each other. Is this thought process correct?

**Comment by lastzero** [comment, marked as answer]
> I use the app "Folder Sync" on my phone to sync the pictures to Nextcloud. I make a folder for each year, so I want the folder 2021 to contain only pictures of 2021, 2020 of 2020 etc. Nextcloud itself doesn't allow for filtering this way, which is why I use Folder Sync.

If PhotoPrism and Nextcloud share the same storage, you may as well upload to PhotoPrism as this should be much faster. Not sure if Nextcloud allows this and/or needs re-indexing though.

> 1. Is this recommended? If not, is there a better way?

Yes, absolutely.

> 2. In Issue #281 I read a few ways of automatic the auto-indexing of originals. I'm not uploading anything via PP, so I have to manually update the index as of now to keep it up2date. I figured I could use cron on my host system or docker. The pics folder is only updated once a day (2am), so I could create a cron tab for 3am to make sure NC and PP don't interfere with each other. Is this thought process correct?

Your pictures will be automatically indexed if you upload to PhotoPrism instead of Nextcloud. Otherwise cron is the way to go for now. We might later add settings for regular tasks.

Hope this helps!

**Comment by FlyingPersian** [comment]
I think it would also be nice to have a filter for auto uploading. E.g. "Starting file name with", so that it would be possible to only upload pictures starting with "IMG_2021".

I think the bigger issue is that nextcloud won't automatically index those new files.
